### PR TITLE
[android] Adds onActivityResult listeners

### PR DESCRIPTION
**Pull request by newnon**
fix crash after wrong getStaticMethodInfo
add some functional to support in app billing(and may be some other extensions)

### CODE REVIEW

**Comment by dumganhar** [comment]
[ci rebuild]

**Comment by newnon** [comment]
Something strange with build server

**Comment by dumganhar** [comment]
@newnon , yes, it seems that there are some errors when pulling codes.
I will rebuild it . Thanks.
[ci rebuild]

**Comment by dumganhar** [comment]
Hi, the title said `small JNI improvement`, I thought it only referred to `JNI`.
So could you separate this PR to two. 
One for `onActivityResult` changes,
Another for `ExceptionClear`. Thanks.

**Comment by newnon** [comment]
done, here is only onActivityResult changes

File: cocos/2d/platform/android/java/src/org/cocos2dx/lib/Cocos2dxHelper.java [hunk 1]
```diff
@@ -60,6 +63,8 @@ of this software and associated documentation files (the "Software"), to deal
 	private static String sFileDirectory;
 	private static Activity sActivity = null;
 	private static Cocos2dxHelperListener sCocos2dxHelperListener;
+	private static Set<OnActivityResultListener> onActivityResultListeners = new LinkedHashSet<OnActivityResultListener>();
+	private static ConcurrentLinkedQueue<Runnable> jobs = new ConcurrentLinkedQueue<Runnable>();
```
Comment (dumganhar) [inline cocos/2d/platform/android/java/src/org/cocos2dx/lib/Cocos2dxHelper.java]: Where did this line used?  `jobs` seems not to be used at all.

**Comment by newnon** [comment]
it was merge error

**Comment by dumganhar** [comment]
Thanks.

**Pull request merged**